src/ng/q.js

```diff
@@ -276,7 +315,7 @@ function qFactory(nextTick, exceptionHandler) {
 
       this.$$state.pending = this.$$state.pending || [];
       this.$$state.pending.push([result, onFulfilled, onRejected, progressBack]);
-      if (this.$$state.status > 0) scheduleProcessQueue(this.$$state);
+      if (this.$$state.status > 0) scheduleProcessQueue(result.promise, this.$$state);
```

Comment:
Here are two scenarios that I would expect to have the same result

```js
// Case 1
var signature = '';
var parent = $qRaf.defer();
var child1 = parent.promise.then(function() {
  signature += 'A';
});
$$rAF(function() {
  var child2 = parent.promise.then(function() {
    signature += 'B';
  });
  parent.resolve('X');
});
$$rAF.flush();
expect(signature).toBe(???); // In this case, signature is 'A'
```

```js
// Case 2
var signature = '';
var parent = $qRaf.defer();
var child1 = parent.promise.then(function() {
  signature += 'A';
});
$$rAF(function() {
  parent.resolve('X');
  var child2 = parent.promise.then(function() {
    signature += 'B';
  });
});
$$rAF.flush();
expect(signature).toBe(???); // In this case, signature is 'AB'
```



To make it fail, I had to chance the implementation of mock.$rAF.flush to only move one frame at a time
```js
  rafFn.flush = function() {
    if (queue.length === 0) {
      throw new Error('No rAF callbacks present');
    }

    var length = queue.length;
    var toFlush = queue;
    queue = [];
    for (var i = 0; i < length; i++) {
      toFlush[i]();
    }
  };
```